src/endpoint/sts/ops/sts_post_assume_role.js

```diff
@@ -49,11 +53,53 @@ async function assume_role(req) {
 }
 
 // create and return the signed token
+/**
+ * @param {Object} auth_options
+ * @param {Number} expiry in seconds
+ * @returns {String}
+*/
 function generate_session_token(auth_options, expiry) {
     dbg.log1('sts_post_assume_role.make_session_token: ', auth_options, expiry);
     return jwt_utils.make_auth_token(auth_options, { expiresIn: expiry });
 }
 
+// TODO: Generalize and move to a utils file in the future
+/**
+ * @param {String|undefined} duration_input
+ * @returns {Number} duration in milliseconds
+ */
+function _parse_sts_duration(duration_input) {
+    if (duration_input === undefined) {
+        return config.STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS;
+    }
+
+    const duration_ms = Number(duration_input) * 1000;
+    const duration_seconds = Math.ceil(duration_ms / 1000);
+
+    if (!Number.isInteger(duration_ms)) {
+        throw new StsError(StsError.InvalidParameterValue);
+    }
```

Comment:
Better check the result of parsing the number before making any arithmetics on it that might change the interpretation of the input. And then I would add the `const duration_ms = duration_sec * 1000` right before the return.

```suggestion
    const duration_sec = Number(duration_input);
    if (!Number.isInteger(duration_sec)) {
        throw new StsError(StsError.InvalidParameterValue);
    }
```